### ISSUE TITLE
Slow speed

### ISSUE THREAD

**Issue by kmchen** [pkg/sftp#36]
Hi 
We were trying to upload a 10MB file to a file on AWS EC2 machines and it takes about 9mins to complete,however, it takes less than 1min with command line on the same machine. Any suggestions that can help to improve the speed? Thanks!!

km

**Comment by davecheney** [comment]
Hello,

I've recently landed #40 which lays the groundwork for interleaving packets which superceeds #24 but does not explicitly issue multiple read and write requests in parallel.

I've also landed #39 which allows the client to specify a larger packet size. Some simple testing shows increasing the size to 64k cuts the benchmark time from 27 seconds to 18 seconds. Increasing the packet larger has mixed results, 256k causes the server to hang up on the machine I tried, or had no increase over the speedup from 64k.

I think this is because the underlying `ssh` package does not support packets larger than 64k so there is minimal benefit in sending larger requests if they are chunked by the underlying ssh layer.

I think these PRs lay a foundation for increasing the throughput, but don't have the bandwidth to take this further at the moment. Hopefully other contributors can assist here.

cc @fdo @Zilog8
